core/src/Kokkos_NumericTraits.hpp

```diff
@@ -141,7 +141,7 @@ template <> struct digits_helper<double> { static constexpr int value = DBL_MANT
 template <> struct digits_helper<long double> { static constexpr int value = LDBL_MANT_DIG; };
 template <class> struct digits10_helper {};
 template <> struct digits10_helper<bool> { static constexpr int value = 0; };
-constexpr double log10_2 = 2.41;
+constexpr double log10_2 = 8*643L/2136; // The fraction 643/2136 approximates log10(2) to 7 significant digits. Follows gcc impl
```

Comment:
Shouldn't this be 
```suggestion
constexpr double log10_2 = 643./2136; // The fraction 643/2136 approximates log10(2) to 7 significant digits. Follows gcc impl
```
? Note both the multiplication with `8`here and making sure the fraction is computed as a floating-point number.

Reply:
I don't believe so.

a) It appears that any floating point in constexpr under `-frounding-math` will trigger that bug, so `constexpr double log10_2 = 643./2136; ` [puts us back to where we started](https://godbolt.org/z/njvbrh5d6). 

b) The result needs to be rounded down, as having 2.41 digits makes no sense. AFAIK `int/int` [rounds to 0](https://stackoverflow.com/a/3602857/410074), which is correct and desired. Indeed I think this `log10_2` should actually be a `constexpr int`, not `double`.

Reply:
Well, your code in the godbolt link shows that what you are proposing is equivalent to `double log10_2=2` which gives wrong results when compared to `std::numeric_limits<>::digits10`. I think I would in the end be OK with just hardcoding the values for `digits10` instead of computing them at compile-time as workaround.